text/0011-validate-requests.md

```diff
@@ -0,0 +1,72 @@
+# Validate input requests
+
+## Related issues and PRs
+
+- Reference Issues:
+- Implementation PR(s):
+
+## Timeline
+
+- Start Date: 2021-06-23
+- Date Entered FCP:
+- Date Accepted:
+- Date Landed:
+
+## Summary
+
+Given a schema, _validation_ checks that policies conform to the schema, and _schema-based parsing_ checks that entities conform to the schema.
+This RFC proposes to add support for checking that a query conforms to a schema.
+
+## Basic example
+
+Say the schema specifies that an action `readFile` should only be used with principals of type `User` and resources of type `File`.
+This RFC proposes to change the signature of the `Request` constructor to add an optional schema argument that can be used to check  argument types when constructing a `Request`.
```

Comment:
Do we have to change the signature of the constructor? Can we add a new constructor instead, with the schema parameter added (whereas the old one stays as is) ?